blog/topology-rules.md

```diff
@@ -0,0 +1,120 @@
+
+{% highlight shell %}
+
+skydive client node-rule create --action="update" --name="update rule" --query="G.V().Has('Name', 'node1')" --metadata="key1=val1, key2=val2"
+{
+  "UUID": "3e6c0e15-a863-4583-6345-715053ac47ce",
+  "Name": "update rule",
+  "Description": "",
+  "Metadata": {
+    "key1": "val1",
+    "key2": "val2"
+  },
+  "Action": "update",
+  "Query": "G.V().Has('Name', 'node1')"
+}
+{% end highlight %}
+
+## Create Edge
+
+To create edge, you should provide the source and destination nodes AND type of edge, the value should be either `layer2` or `ownership`.
+
+{% highlight shell %}
+skydive client edge-rule create --name="edge" --src="G.v().has('TID', '2f6f9b99-82ef-5507-76b6-cbab28bda9cb')" --dst="G.V().Has('TID', 'd6ec6e2f-362e-51e5-4bb5-6ade37c2ca5c')" --relationtype="both"
+{
+  "UUID": "50fec124-c6d0-40c7-42a3-2ed8d5fbd410",
+  "Name": "edge",
+  "Description": "",
+  "Src": "G.v().has('TID', '2f6f9b99-82ef-5507-76b6-cbab28bda9cb')",
+  "Dst": "G.V().Has('TID', 'd6ec6e2f-362e-51e5-4bb5-6ade37c2ca5c')",
+  "Metadata": {
+    "RelationType": "both"
+  }
+}
+{% end highlight %}
+
+## USECASE 1
+
+In this use case, we will see how to add a network device which is not part of the skydive network. Let’s consider this example, we have two skydive agents running in two different hosts, to connect those two hosts we need a TOR switch. Even Though we can achieve this via defining a fabric nodes and links in config file, let’s see how we can do the same using Topology rules API.
+
+Without TOR switch, two agents will look like two different node without any links as shown in the below image.
+
+<p class="center">
+  <a href="/assets/images/blog/rules1.png" data-lightbox="Skydive Topology" data-title="Skydive topology view">
+    <img src="/assests/images/blog/rules1.png"/>
+  </a>
+</p>
+
+Now run the below Node Rule commands to create a TOR switch and Ports.
+
+{% highlight shell %}
+skydive client node-rule create --node-name="TOR" --node-type="fabric" --action="create"
+skydive client node-rule create --node-name="port1" --node-type="port" --action="create"
+skydive client node-rule create --node-name="port2" --node-type="port" --action="create"
+{% end highlight %}
+
+You can see, the TOR switch and ports are created and added to the skydive topology and now the topology will look like as shown in the below image.
+
+<p class="center">
+  <a href="/assets/images/blog/rules2.png" data-lightbox="Skydive Topology" data-title="Skydive topology view">
+    <img src="/assests/images/blog/rules2.png"/>
+  </a>
+</p>
+
+Now run the below Edge Rule commands to create a links between the switch, ports and the public interfaces of the hosts.
+
+{% hghlight shell %}
+skydive client edge-rule create --src="G.V().Has('Name', 'TOR')" --dst="G.V().Has('Name', 'port1')" --relationtype="ownership"
```

Comment:
I would split this in two parts like TOR -> port1 and another highlight section for TOR -> port2, It will be better for readability. I would use "\" in the middle to break the line in two parts.

Reply:
done

Reply:
done